awx/api/filters.py

```diff
@@ -70,7 +70,7 @@ def filter_queryset(self, request, queryset, view):
                     types_map[ct_type] = ct.pk
                 model = queryset.model
                 model_type = get_type_for_model(model)
-                if 'polymorphic_ctype' in model._meta.get_all_field_names():
+                if 'polymorphic_ctype' in [f.name for f in model._meta.get_fields()]:
```

Comment:
get_fields doesn't include m2m and many-related fields, did the previous one include this?

Reply:
actually, I think I wasn't exactly right, but still

```
In [3]: len(Credential._meta.get_all_field_names())
Out[3]: 35

In [4]: len(Credential._meta.get_fields())
Out[4]: 28
```

Reply:
Based on what is here, https://docs.djangoproject.com/en/1.11/ref/models/meta/#migrating-from-the-old-api

I would of expected my version to return more field names than the old version, since the `get_all _field_names` filtered fields associated with GFK.

Reply:
```
MyModel._meta.get_all_field_names() becomes:

from itertools import chain
list(set(chain.from_iterable(
    (field.name, field.attname) if hasattr(field, 'attname') else (field.name,)
    for field in MyModel._meta.get_fields()
    # For complete backwards compatibility, you may want to exclude
    # GenericForeignKey from the results.
    if not (field.many_to_one and field.related_model is None)
)))
This provides a 100% backwards compatible replacement, ensuring that both field names and attribute names ForeignKeys are included, but fields associated with GenericForeignKeys are not. A simpler version would be:

[f.name for f in MyModel._meta.get_fields()]
While this isn’t 100% backwards compatible, it may be sufficient in many situations.
```

Reply:
I see now using `chain.from_iterable` grabs the nested lists where just a list comp does not. I'll update it.